editor/code/src/client.ts

```diff
@@ -431,9 +442,15 @@ export function activateCoqLSP(
   // We violate this on the error case, but only because it is exceptional.
   const updateStatusBar = () => {
     if (client && client.isRunning()) {
-      lspStatusItem.text = "$(check) coq-lsp (running)";
-      lspStatusItem.backgroundColor = undefined;
-      lspStatusItem.tooltip = "coq-lsp is running. Click to disable.";
+      if (serverConfig.check_only_on_request) {
+        lspStatusItem.text = "$(check) coq-lsp (on-demand checking)";
+        lspStatusItem.backgroundColor = undefined;
+        lspStatusItem.tooltip = "coq-lsp is running. Click to disable.";
+      } else {
+        lspStatusItem.text = "$(check) coq-lsp (continous checking)";
```

Comment:
I think this button should only appear when the language server is enabled via the other button.

Reply:
What is the "other button"?

But yes, I think we should move this to its own language server status

Reply:
The other button is the enable/disable button that we already have. It doesn't make sense for the coq-lsp to be disabled and this button visible, IMO.

Reply:
This is the same button, isn't it?

Reply:
Oh then I got confused. I thought you added another button when I read it this morning. Nevermind.

Reply:
For now we have a single button with 4 states (enabled (continous/lazy) / disabled / failed), but I think we should move this stuff to a languageStatusItem actually, I will implement a `serverStatus` notification and do this, will look nice actually.